src/libraries/System.Private.CoreLib/src/System/Numerics/Vector.cs

```diff
@@ -1431,5 +1433,27 @@ internal static void ThrowInsufficientNumberOfElementsException(int requiredElem
         {
             throw new IndexOutOfRangeException(SR.Format(SR.Arg_InsufficientNumberOfElements, requiredElementCount, "values"));
         }
+
+        /// <summary>
+        /// Reinterprets a <see cref="Vector{T}"/> as a <see cref="Vector{T}"/> of new type.
+        /// </summary>
+        /// <typeparam name="TFrom">The type of the input vector.</typeparam>
+        /// <typeparam name="TTo">The type to reinterpret the vector as.</typeparam>
+        /// <param name="vector">The vector to reinterpret.</param>
+        /// <returns><paramref name="vector"/> reinterpreted as a new <see cref="Vector{T}"/>.</returns>
+        /// <exception cref="NotSupportedException">
+        /// The type of <typeparamref name="TFrom"/> or <typeparamref name="TTo"/> is not supported.
+        /// </exception>
+        public static Vector<TTo> As<TFrom, TTo>(Vector<TFrom> vector)
```

Comment:
The runtime support for this is actually not that complicated. Basically:
1. Add the `[Intrinsic]` attribute to this method
2. Add an entry for the method in the "SIMD as HWIntrinsic" metadata tables
  A. For x86 this is adding an entry for `As` to https://github.com/dotnet/runtime/blob/master/src/coreclr/jit/simdashwintrinsiclistxarch.h#L104 and https://github.com/dotnet/runtime/blob/master/src/coreclr/jit/simdashwintrinsiclistxarch.h#L140
  B. For ARM This is adding an entry to https://github.com/dotnet/runtime/blob/master/src/coreclr/jit/simdashwintrinsiclistarm64.h#L104
  C. You should be able to have the name for all 10 `TYP_` slots forward to the method of the same name (`NI_VectorT128_As` or `NI_VectorT256_As`, respectively)
3. Update the line here to include `NI_VectorT128_As` as `isOpExplicit` and the line just below to include `NI_VectorT256_As`: https://github.com/dotnet/runtime/blob/master/src/coreclr/jit/simdashwintrinsic.cpp#L566

Since this is a `static` method on `Vector`, it should validate `TFrom` here: https://github.com/dotnet/runtime/blob/master/src/coreclr/jit/simdashwintrinsic.cpp#L246
I believe you will then also need to add a couple lines to this switch statement checking the return type: https://github.com/dotnet/runtime/blob/master/src/coreclr/jit/simdashwintrinsic.cpp#L391
The logic would be essentially:
```cpp
#if defined(TARGET_X86)
case NI_VectorT256_As:
{
    FALLTHROUGH;
}
#endif // TARGET_X86

case NI_VectorT128_As:
{
    unsigned retSimdSize;
    var_types retBaseType = getBaseTypeAndSizeOfSIMDType(sig->retTypeSigClass, &retSimdSize);

    if (!varTypeIsArithmetic(retBaseType) || (retSimdSize == 0))
    {
        // We get here if the return type is an unsupported type
        return nullptr;
    }
    break;
}

Reply:
Will address this tomorrow